optimum/intel/openvino/modeling_timm.py

```diff
@@ -0,0 +1,303 @@
+import os
+from collections import OrderedDict
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import timm
+import torch
+from timm.layers.config import set_fused_attn
+from timm.models._hub import load_model_config_from_hf
+from transformers import PretrainedConfig, PreTrainedModel
+from transformers.image_processing_utils import BaseImageProcessor, BatchFeature, get_size_dict
+from transformers.image_transforms import resize, to_channel_dimension_format
+from transformers.image_utils import (
+    IMAGENET_STANDARD_MEAN,
+    IMAGENET_STANDARD_STD,
+    ChannelDimension,
+    ImageFeatureExtractionMixin,
+    ImageInput,
+    PILImageResampling,
+    make_list_of_images,
+    to_numpy_array,
+    valid_images,
+)
+from transformers.modeling_outputs import ImageClassifierOutput
+from transformers.utils import TensorType
+
+from optimum.exporters.onnx.model_configs import ViTOnnxConfig
+
+
+set_fused_attn(False, False)
+
+
+class TimmConfig(PretrainedConfig):
+    model_type = "timm"
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        pretrained_model_name_or_path: Union[str, os.PathLike],
+        cache_dir: Optional[Union[str, os.PathLike]] = None,
+        force_download: bool = False,
+        local_files_only: bool = False,
+        token: Optional[Union[str, bool]] = None,
+        revision: str = "main",
+        **kwargs,
+    ) -> "PretrainedConfig":
+        kwargs["cache_dir"] = cache_dir
+        kwargs["force_download"] = force_download
+        kwargs["local_files_only"] = local_files_only
+        kwargs["revision"] = revision
+
+        config_dict = load_model_config_from_hf(pretrained_model_name_or_path)[0]
+        config_dict["num_labels"] = config_dict.pop("num_classes")
+        config_dict["image_size"] = config_dict.get("input_size")[-1]
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class TimmOnnxConfig(ViTOnnxConfig):
```

Comment:
You used `ViTOnnxConfig` as a base class here. Does it mean that only ViT models from Timm will work here?

Reply:
I would inherit from `VisionOnnxConfig` to make it safer.